### PR TITLE
[Tests-Only]Add tests for testContext

### PR DESCRIPTION
**Pull request by jasson99**
## Description:
This PR adds unit tests for testContext.

## Related Issue:
- https://github.com/owncloud/owncloud-test-middleware/issues/15

### CODE REVIEW

File: src/gherkin/testContext.test.js [hunk 1]
```diff
@@ -0,0 +1,261 @@
+/* eslint no-undef: 0 */
+const { Step } = require("./index");
+const { TestContext } = require("./index");
+const { StepDef } = require("./index");
```
Comment (dpakach) [inline src/gherkin/testContext.test.js:4]:
```suggestion
const { Step, TestContext, StepDef } = require("./index");
```

File: src/gherkin/testContext.test.js [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    expect(testContext.steps).toStrictEqual(expectedStep);
+  });
+
+  it("test before and after function", () => {
+    const testContext = new TestContext();
+    testContext.before("abc");
+    expect(testContext.beforeSteps).toStrictEqual(["abc"]);
+    testContext.after("xyz");
+    expect(testContext.afterSteps).toStrictEqual(["xyz"]);
+  });
```
Comment (dpakach) [inline src/gherkin/testContext.test.js:260]: before and after should only expect functions, you should check that also test calling setup() and cleanup()

File: src/gherkin/testContext.test.js (hunk 1 above)
Comment (jasson99) [reply src/gherkin/testContext.test.js]: fixed

**Review by dpakach — CHANGES_REQUESTED** [review]
Looks good please squash your commits

File: src/gherkin/testContext.test.js (hunk 2 above)
Comment (jasson99) [reply src/gherkin/testContext.test.js]: fixed

**Comment by individual-it** [comment]
Do we even need WHEN steps here?

**Comment by dpakach** [comment]
@individual-it I don't think we will be using any when steps from the middleware but it doesn't hurt to just leave it here IMO.

**Review by individual-it — APPROVED** [review]

**Review by dpakach — APPROVED** [review]

**Pull request merged**